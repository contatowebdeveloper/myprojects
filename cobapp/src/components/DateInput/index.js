import React, { useState, useEffect } from 'react'
import DatePicker, { registerLocale  } from 'react-datepicker'
import ptbr from 'date-fns/locale/pt-BR'
import 'react-datepicker/dist/react-datepicker.css'
import MaskedTextInput from 'react-text-mask'

registerLocale("ptbr", ptbr)

export default function DateInput(props) {
    const [ startDate, setStartDate ] = useState('')
    const [ enabledClear, setEnabledClear ] = useState(false)
    const { placeholder, clearValueDate } = props

    useEffect(() => {
        if(clearValueDate){
            setStartDate('')
            setEnabledClear(false)
        }
    }, [clearValueDate, props]);

    function handleChange(date) {
        
		if(date){
            setStartDate(date)
            setEnabledClear(true)
		} else {
            setStartDate('')
            setEnabledClear(false)
		}
    }

    return(
        <div className="DateCob-input">				
            <DatePicker
                id={props.id}
                isClearable={enabledClear}
                placeholderText={placeholder}
                selected={startDate}
                onChange={handleChange}
                className="DateCob-picker"
                dateFormat="dd/MM/yyyy"
                locale="ptbr"
                customInput={
                    <MaskedTextInput
                        type="text"
                        mask={[/\d/, /\d/, "/", /\d/, /\d/, "/", /\d/, /\d/, /\d/, /\d/]}
                    />
                }
            />			
        </div>
    )
}