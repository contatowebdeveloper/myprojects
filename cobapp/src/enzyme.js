import Enzyme, { configure, shallow, mount, render} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter : new Adapter() })
export { shallow, mount, render }
export default Enzyme

/***** FUNÇÕES DO ENZYME *****/
//função shallow serve para renderizar o componente mas sem incluir os componentes filhos e LifeCycles.
//função render serve para renderizar o componente incluindo os componente filhos mas não executa seus LifeCycles, apenas o método render do componente.
//função Mount serve para renderizar o componente incluindo os componentes filhos e executar seus LifeCycles.
//converte as funções do Enzyme para funcionar com os testes de snapshot do Jest.