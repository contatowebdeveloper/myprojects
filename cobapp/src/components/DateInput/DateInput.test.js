import React from 'react'
import { shallow, mount } from '../../enzyme'
import TestRenderer from 'react-test-renderer'

import DateInput from './index'

jest.mock('react-text-mask', () => props => <input type="text" {...{ ...props }} />)

describe('Render DateInput component', () => {
    it('render correctly date component and create a snapshot', () => {
        const wrapper = shallow(<DateInput />)
        expect(wrapper.find('DatePicker').length).toBe(1)
        const DateInputComponent = TestRenderer.create(<DateInput />).toJSON()
        expect(DateInputComponent).toMatchSnapshot()
    })

    it('testing dateInput passed a string', () => {
        const props = {
            value : '10.05.2018'
        }
        const DateInputComponent = mount(<DateInput {...props} />)
        expect(DateInputComponent.prop('value')).toBe('10.05.2018')
    })

    it('render date input correctly with null value', () => {  
        const props = { 
            value: null 
        }
        const DateInputComponent = mount(<DateInput {...props} />)
        expect((DateInputComponent).prop('value')).toBeNull()
    })

    it('testing dateInput passed a number and expected a string', () => {
        const props = {
            value : '10/10/2019'
        }
        const DateInputComponent = mount(<DateInput {...props} />)
        expect(DateInputComponent.prop('value')).toEqual('10/10/2019') //aqui Ã© pra falhar
    })
})