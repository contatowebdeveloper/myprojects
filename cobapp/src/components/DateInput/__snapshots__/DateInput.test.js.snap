// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render DateInput component render correctly date component and create a snapshot 1`] = `
<div
  className="DateCob-input"
>
  <div
    className="react-datepicker-wrapper"
  >
    <div
      className="react-datepicker__input-container"
    >
      <input
        className="DateCob-picker"
        disabled={false}
        mask={
          Array [
            /\\\\d/,
            /\\\\d/,
            "/",
            /\\\\d/,
            /\\\\d/,
            "/",
            /\\\\d/,
            /\\\\d/,
            /\\\\d/,
            /\\\\d/,
          ]
        }
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        readOnly={false}
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
